File: JS-Solutions/Problem-06.js
// Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum

const sumOfSquares = (limit) => {
  let sum = 0;
  for (let i=1;i<=limit;i++) {
    sum += i**2;
  }
  return sum;
}

const squareOfSum = (limit) => {
  let sum = 0;
  for (let i=1;i<=limit;i++) {
    sum += i;
  }
  return sum**2;
}

const squareMinusSum = (limit) => {
  return squareOfSum(limit) - sumOfSquares(limit);
}

console.log(squareMinusSum(100));

File: Swift-Solutions/Problem-06.swift
// Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.
import Foundation

func squareOfSum(limit: Int) -> Int {
  var sum: Int = 0
  for i in 1...limit {
    sum += i
  }
  return Int(pow(Double(sum),2))
}

func sumOfSquares(limit: Int) -> Int {
  var sum: Double = 0
  for i in 1...limit {
    sum += pow(Double(i),2)
  }
  return Int(sum)
}

func squareMinusSum(limit: Int) -> Int {
  return squareOfSum(limit: limit) - sumOfSquares(limit: limit)
}

print(squareMinusSum(limit: 100))
